nixos/modules/programs/lomiri.nix

```diff
@@ -0,0 +1,160 @@
+{ config, pkgs, lib, ... }:
+
+let
+  cfg = config.programs.lomiri;
+in {
+  options.programs.lomiri= {
+    enable = lib.mkEnableOption (lib.mdDoc ''
+      the Lomiri graphical shell (formerly known as Unity8)'');
+  };
+
+  config = lib.mkIf cfg.enable {
+    environment = {
+      systemPackages = (with pkgs; [
+        # Required/Expected user services
+        libayatana-common
+        ubports-click
+
+        # Data
+        vanilla-dmz # TODO is this used in Lomiri?
+      ]) ++ (with pkgs.lomiri; [
+        lomiri-session # Wrappers to properly launch the session
+        lomiri
+        qtmir # not having its desktop file for Xwayland available causes any X11 application to crash the session
+        lomiri-system-settings
+        morph-browser
+        lomiri-terminal-app
+
+        libusermetrics
+
+        # TODO OSK does not work yet
+        # lomiri-keyboard is a plugin for maliit-framework, which still seems incapable of loading any plugins at all.
+        # maliit-framework's maliit-server hardcodes a plugin name & location to load, and needs the plugin's glib schema.
+        # It (maliit-server or the plugin?) also has a hardcoded path to hunspell dictionaries for text prediction
+        #maliit-framework
+        #lomiri-keyboard
+
+        # Required/Expected user services
+        lomiri-thumbnailer
+        lomiri-url-dispatcher
+        lomiri-download-manager
+        lomiri-schemas # exposes some required dbus interfaces
+        hfd-service
+        history-service
+        telephony-service
+        content-hub
+        mediascanner2 # possibly needs to be kicked off by graphical-session.target
+
+        lomiri-sounds
+        lomiri-wallpapers
+        suru-icon-theme
+      ]);
+    };
+
+    # Required/Expected system services
+    systemd.packages = with pkgs.lomiri; [
+      hfd-service
+      lomiri-download-manager
+    ];
+
+    services.dbus.packages = with pkgs.lomiri; [
+      hfd-service
+      libusermetrics
+      lomiri-download-manager
+    ];
+
+    fonts.packages = with pkgs; [
+      # Applications tend to default to Ubuntu font
+      ubuntu_font_family
+    ];
+
+    # Copy-pasted
+    # TODO are all of these needed? just nice-have's? convenience?
+    hardware.opengl.enable = lib.mkDefault true;
+    fonts.enableDefaultPackages = lib.mkDefault true;
+    programs.dconf.enable = lib.mkDefault true;
+    programs.xwayland.enable = lib.mkDefault true;
```

Comment:
I think those should become part of a shared module between xserver and wayland envs. After splitting out xserver this is on my todo

Reply:
Agree, will keep them copy-pasted here until then.